contracts/MANACrowdsale.sol

```diff
@@ -82,7 +86,11 @@ contract MANACrowdsale is WhitelistedCrowdsale, CappedCrowdsale, FinalizableCrow
         }
 
         // otherwise compute the price for the auction
-        return rate.sub(rateStepDecrease.mul(block.number - startBlock));
+        uint256 elapsed = block.number - startBlock;
+        uint256 rateRange = initialRate - endRate;
+        uint256 blockRange = endBlock - startBlock;
+
+        return initialRate.sub(rateRange.div(blockRange).mul(elapsed));
```

Comment:
Should be `rateRange.mul(elapsed).div(blockRange)`, right? Otherwise `rateRange.div(blockRange)` is less than 1, ends up being zero, and we are in the same spot as before.